content/blog/20241126-100plus-gigabit-ethernet-in-fpga.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "100+ Gigabit Ethernet in FPGA"
+date: "2024-11-25"
+author: "Jakub Cabal"
+tags: 
+  - "ethernet"
+  - "hard-ip"
+---
+
+Ethernet is the technology that allows us to connect individual devices to networks and the entire Internet. 
+Different devices use Ethernet for communication, remote control and also for data transfer.
+It is therefore no coincidence that Ethernet support is also found in a wide range of FPGAs.
+
+Implementing 100 Mbps Ethernet support is not difficult even on the cheapest FPGAs,
+but higher speeds may require more sophisticated hardware.
+
+## There is no Ethernet like Ethernet
+
+Under the word Ethernet there are also many standards that differ not only in transmission speed.
+Some standards are designed for metallic cables, others for fiber optics.
+For example, 100 Gigabit Ethernet can be transmitted over four serial lines or just two lines.
+The difference may also be that some standards require RS-FEC (Reed–Solomon error correction, there are different variants) and others do not.
+
+There are a lot of these Ethernet standards,
+for an example you can look at the [list of just those for 100 Gbps transmission speeds](https://en.wikipedia.org/wiki/100_Gigabit_Ethernet#100G_interface_types).
+The engineer should know exactly what he will be working with, but it is not easy to know.
+Fortunately, there are only a few widely used Ethernet standards and we don't see many others in practice.
+
+## Multi Gigabit Ethernet support may vary
+
+Ethernet support in FPGAs can vary not only in transmission speed,
+but also the level of built-in support (within the Hard IP blocks).
+
+Some FPGAs have only fast enough serial lines (SERDES or also referred to as transceivers) to support proper encoding (NRZ, PAM4, etc.),
+and therefore allow to implement PMA layer Ethernet support.
+Better FPGAs have built-in logic to decode the Ethernet protocol at PCS level to the [Media-Independent Interface (MII)](https://en.wikipedia.org/wiki/Media-independent_interface) interface.
+However, the main trend is to provide full support, and so the best FPGAs have dedicated logic for all key Ethernet protocol layers including MAC and sometimes optional RS-FEC layer.
+
+**So once again, briefly to summarize:**
+
+- PMA - support only at the high-speed serial lines (SERDES) level
+- PMA+PCS - support including decoding to MII interface level
+- PMA+PCS+MAC+(RS-FEC) - full Ethernet protocol support
+
+If your FPGA has support for at least the PAM layer, you can implement the remaining ones in the gate array.
+Such solutions (in the form of Soft IP) are provided by FPGA chip manufacturers directly or by other companies for a considerable licensing fee.
+
+## Compare 100+ Gigabit Ethernet Hard IPs
+
+Among FPGA chip manufacturers, AMD (Xilinx) and Altera (Intel) are the two main companies dominating.
```

Comment:
To "dominating" na konci vety redundantné. Alebo by som prehodil slovosled na "two main dominating companies". Ale skôr by si vybral buď main alebo dominating.